code/drasil-gool/lib/Drasil/Shared/InterfaceCommon.hs

```diff
@@ -89,6 +89,8 @@ class TypeSym r where
   funcType      :: [VSType r] -> VSType r -> VSType r
   void          :: VSType r
 
+-- We've established that getTypeString is a bad thing
+-- getType might be required, but without global state should it be a class?
```

Comment:
`getType` is used to do a little bit of introspection. There might be other ways to do that.

It's a class because we should always be able to do that, and implementation can be per-language.